### PR TITLE
[Logs]: Added a support for log-collection

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Updated the readme to setup log-collection and added the possibility to collect all logs from `STDOUT`, `STDERR`

### Motivation

Collect logs with the puppy agent

### CODE REVIEW

File: lib/run-datadog.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ start_datadog() {
     export DD_LOG_FILE=$DATADOG_DIR/dogstatsd.log
     export DD_API_KEY
     export DD_DD_URL=${DD_DD_URL:-https://app.datadoghq.com}
-    export DD_ENABLE_CHECKS="${DD_ENABLE_CHECKS:-true}"
+    export DD_ENABLE_CHECKS="${DD_ENABLE_CHECKS:-false}"
```
Comment (arbll) [inline lib/run-datadog.sh]: Unfortunately we can't do that as it would impact existing installs

File: lib/run-datadog.sh [hunk 2]
```diff
@@ -64,8 +64,9 @@ start_datadog() {
 
 if [ -z $DD_API_KEY ]; then
   echo "Datadog API Key not set, not starting Datadog"
+elif [ "$DD_LOGS_ENABLED" = "true" -a -z $DD_LOGS_CONFIG_TCP_FORWARD_PORT ]; then
+  echo "TCP forward port is not set, not starting Datadog"
```
Comment (tmichelet) [inline lib/run-datadog.sh]: We could have a better message here, for instance `Logs are enabled but not properly setup: missing TCP forwarding port option.

Also, I think we should still start the agent in that case, but may be wrong.

File: README.md [hunk 3]
```diff
@@ -24,6 +24,19 @@ cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
...
 ```
 
+You can also enable log collection, you will need to provide a TCP port to forward logs to the agent:
+```shell
+# set the environment variables
+cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
+cf set-env $YOUR_APP_NAME RUN_PUPPY true
+cf set-env $YOUR_APP_NAME DD_LOGS_ENABLED true
+cf set-env $YOUR_APP_NAME DD_LOGS_CONFIG_TCP_FORWARD_PORT 10514
+# prevent collecting all logs from stdout
```
Comment (tmichelet) [inline README.md]: could we improve the comment? Like explain what is the default behavior and how setting it to false will change it?

File: lib/run-datadog.sh (hunk 2 above)
Comment (ajacquemot) [reply lib/run-datadog.sh]: Moved this to `redirect_logs.sh`

File: README.md [hunk 4]
```diff
@@ -24,6 +24,21 @@ cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
...
 
+You can also enable log collection, you will need to provide a TCP port to forward logs to the agent.
+By default, it collects all logs on stdout and stderr, you can disable this functionality if you just want your application logs,
+you will need to forward your logs to the agent using TCP forwarding.
+```shell
+# set the environment variables
+cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
+cf set-env $YOUR_APP_NAME RUN_PUPPY true
+cf set-env $YOUR_APP_NAME DD_LOGS_ENABLED true
+cf set-env $YOUR_APP_NAME DD_LOGS_CONFIG_TCP_FORWARD_PORT 10514
```
Comment (arbll) [inline README.md]: The purpose of this option might be a bit confusing for the users. I think we should inform them that they just need to put an unused TCP port

Comment (ajacquemot) [reply README.md]: it's always used, see https://github.com/DataDog/datadog-cloudfoundry-buildpack/blob/b22bdb5d343666babf260ef9b0d819d8e3ad22cf/lib/redirect-logs.sh#L5

Comment (arbll) [reply README.md]: Yes I get that, I meant as someone that only want to forward the logs from my std(out|err) "why do I have to configure a port ?"

Comment (arbll) [reply README.md]: But that was just a thought, feel free to ignore ;)

Comment (ajacquemot) [reply README.md]: That's totally valid, we could set a default value.

Comment (ajacquemot) [reply README.md]: Updated

**Pull request merged**